src/main/app/access-code/AccessCodePostController.ts

```diff
@@ -0,0 +1,57 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { getNextStepUrl } from '../../steps';
+import { getCaseWorkerUser } from '../auth/user/oidc';
+import { getCaseApi } from '../case/CaseApi';
+import { CITIZEN_LINK_APPLICANT_2 } from '../case/definition';
+import { AppRequest } from '../controller/AppRequest';
+import { AnyObject, PostController } from '../controller/PostController';
+import { Form } from '../form/Form';
+
+@autobind
+export class AccessCodePostController extends PostController<AnyObject> {
+  constructor(protected readonly form: Form) {
+    super(form);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const caseworkerUser = await getCaseWorkerUser();
+    req.locals.api = getCaseApi(caseworkerUser, req.locals.logger);
```

Comment:
My preference would not to be set this in locals as it's hard to know what else uses it and we want to make absolutely sure this API is only used to load and link the case. It would require changing `this.save(...)` below to call the event in the API manually, but I don't know why you haven't done that anyway - is there something in `this.save(...)` you need?

Reply:
this.save() is just the save in the PostController

Reply:
yeah that's what I mean - is there a reason you need to use that, you could just trigger the event on the api